### ISSUE TITLE
GDScript 2.0: Packed arrays are not considered constant expressions

### ISSUE THREAD

**Issue by dalexeev** [godotengine/godot#67873]
### Godot version

v4.0.beta3.official [01ae26d31]

### System information

Kubuntu 22.10

### Issue description

Attempting to use a packed array in a constant definition causes "Initializer for a constant must be a constant expression" error.

### Steps to reproduce

```gdscript
# 3.x:
const A = PoolIntArray() # OK.
const B = PoolIntArray([1, 2, 3]) # OK.

# 4.0-dev:
const A = PackedInt32Array() # Error: Initializer for a constant must be a constant expression.
const B = PackedInt32Array([1, 2, 3]) # Error: Initializer for a constant must be a constant expression.
```

**Comment by dalexeev** [comment]
Also, we should not forget about readonly `Packed*Array`s (`make_read_only()`), for consistency with `Array` and `Dictionary`.

**Comment by dalexeev** [comment]
> This might be tricky because `Packed*Array` are just `typedef`s for `Vector<>`. Adding read-only functionality would affect all vectors used in core.

Yes, I was worried about it, and that's why I tried to draw attention to it yesterday. :)

**Comment by dalexeev** [comment]
I suggest deferring this to 4.2 or 4.x as 4.1 is in a feature freeze. Since this is more a lack of a new feature than a bug. Packed arrays have changed significantly in 4.0, they are now passed by reference rather than by value. And since they don't have a `make_read_only()` method (unlike `Array` and `Dictionary`), we can't make them constant as it worked before on the GDScript side. Core changes are required, and now is not the time to experiment with such important engine structure as `Vector<T>`. Please add the `topic:core` label.

**Comment by Calinou** [comment]
I can still reproduce this on 4.5.dev 215acd52e:

```gdscript
const c = PackedInt32Array()  # Assigned value for constant "c" isn't a constant expression.
```

**Comment by DevPoodle** [comment]
For anyone trying to statically type a const packed array, although this doesn't work:

``` gdscript
const ARRAY := PackedInt32Array([1, 2, 3])
```
This does:

``` gdscript
const ARRAY : PackedInt32Array = [1, 2, 3]
```
